### PR TITLE
add sorting for candidate quizzes

### PR DESCRIPTION
**Pull request by BaiasPaul**
add sorting for candidate quizzes
Jira: https://jira.evozon.com/browse/PHPTRAININ-265

### CODE REVIEW

File: src/Controller/QuizInstanceController.php [hunk 1]
```diff
@@ -56,16 +66,22 @@ public function showCandidateQuizzes(Request $request, array $requestAttributes)
         if ($redirectToLogin) {
             return $redirectToLogin;
         }
+        $orderBy = $this->quizInstanceService->getFromParameter('orderBy', $request, "");
+        $sortType = $this->quizInstanceService->getFromParameter('sort', $request, "");
         //TODO remove cast and modify methods to return the expected type
         $currentPage = (int)$this->quizInstanceService->getFromParameter('page', $request, 1);
         $totalResults = (int)$this->quizInstanceService->getEntityNumberOfPagesByField(QuizTemplate::class, []);
```
Comment (victortodoran) [inline src/Controller/QuizInstanceController.php:81]: Law of Demeter.

File: src/Controller/QuizInstanceController.php [hunk 2]
```diff
@@ -56,16 +66,22 @@ public function showCandidateQuizzes(Request $request, array $requestAttributes)
         if ($redirectToLogin) {
             return $redirectToLogin;
         }
```
Comment (victortodoran) [inline src/Controller/QuizInstanceController.php:72]: Extract logic that builds filter into separate method.

File: src/Controller/QuizInstanceController.php [hunk 3]
```diff
@@ -56,16 +66,22 @@ public function showCandidateQuizzes(Request $request, array $requestAttributes)
...
+        $orderBy = $this->quizInstanceService->getFromParameter('orderBy', $request, "");
+        $sortType = $this->quizInstanceService->getFromParameter('sort', $request, "");
         //TODO remove cast and modify methods to return the expected type
         $currentPage = (int)$this->quizInstanceService->getFromParameter('page', $request, 1);
         $totalResults = (int)$this->quizInstanceService->getEntityNumberOfPagesByField(QuizTemplate::class, []);
-        $quizzes = $this->quizInstanceService->getEntitiesByField(QuizTemplate::class, [], $currentPage, self::RESULTS_PER_PAGE);
+        $quizzes = $this->repoManager->getRepository(QuizTemplate::class)->getFilteredEntities(
+            new Filter([], self::RESULTS_PER_PAGE, ($currentPage - 1) * self::RESULTS_PER_PAGE, $orderBy, $sortType)
+        );
         $paginator = new Paginator($totalResults, $currentPage, self::RESULTS_PER_PAGE);
```
Comment (victortodoran) [inline src/Controller/QuizInstanceController.php:93]: why not pass the filter to the template? It already has the orderBy and sortType grouped together.

File: views/candidate-quiz-listing.phtml [hunk 4]
```diff
@@ -52,7 +52,7 @@
         <table class="table table-hover my-4">
             <thead>
             <tr>
```
Comment (victortodoran) [inline views/candidate-quiz-listing.phtml:54]: too much logic to follow in this <th> please make it easier to understand.

File: src/Controller/QuizInstanceController.php [hunk 5]
```diff
@@ -56,16 +79,33 @@ public function showCandidateQuizzes(Request $request, array $requestAttributes)
...
+        );
+
+        $quizzes = $this->repoManager->getRepository(QuizTemplate::class)->getFilteredEntities($filtersForEntity);
         $paginator = new Paginator($totalResults, $currentPage, self::RESULTS_PER_PAGE);
 
         return $this->renderer->renderView("candidate-quiz-listing.phtml", [
             'username' => $this->quizInstanceService->getName(),
             'paginator' => $paginator,
             'quizzes' => $quizzes,
+            'url' => $this->urlBuilder,
```
Comment (calinpristavu) [inline src/Controller/QuizInstanceController.php]: inject UrlBuilder from the framework into all templates

**Review by calinpristavu — CHANGES_REQUESTED** [review]

**Pull request merged**